extensions/ql-vscode/src/databases.ts

```diff
@@ -525,7 +532,15 @@ export class DatabaseManager extends DisposableObject {
     token: vscode.CancellationToken,
     uri: vscode.Uri,
   ): Promise<DatabaseItem> {
-    const contents = await resolveDatabaseContents(uri);
+    let contents = await resolveDatabaseContents(uri);
+    const managedUri = await this.getDatabaseInManagedLocation(uri);
+    if (managedUri !== uri) {
+      // We always want to run resolveDatabaseContents() before doing any substantial work so that
+      // we can report obvious problems to the user quickly. If we end up copying the database
+      // directory to a new location, then we need to run resolveDatabaseContents() again to get
+      // the correct contents for the copied database directory.
+      contents = await resolveDatabaseContents(managedUri);
+    }
```

Comment:
Is there a reason to resolve database contents twice?

```suggestion
    let contents;
    const managedUri = await this.getDatabaseInManagedLocation(uri);
    if (managedUri !== uri) {
      // We always want to run resolveDatabaseContents() before doing any substantial work so that
      // we can report obvious problems to the user quickly. If we end up copying the database
      // directory to a new location, then we need to run resolveDatabaseContents() again to get
      // the correct contents for the copied database directory.
      contents = await resolveDatabaseContents(managedUri);
    } else {
      contents = await resolveDatabaseContents(uri);
    }
```